farm/modeling/adaptive_model.py

```diff
@@ -818,13 +725,21 @@ def forward(self, **kwargs):
         :return: all logits as torch.tensor or multiple tensors.
         """
         with torch.no_grad():
-            input_to_onnx = {
-                'input_ids': numpy.ascontiguousarray(kwargs['input_ids'].cpu().numpy()),
-                'padding_mask': numpy.ascontiguousarray(kwargs['padding_mask'].cpu().numpy()),
-                'segment_ids': numpy.ascontiguousarray(kwargs['segment_ids'].cpu().numpy()),
-            }
+            if self.language_model_class == "Bert":
+                input_to_onnx = {
+                    'input_ids': numpy.ascontiguousarray(kwargs['input_ids'].cpu().numpy()),
+                    'attention_mask': numpy.ascontiguousarray(kwargs['padding_mask'].cpu().numpy()),
+                    'token_type_ids': numpy.ascontiguousarray(kwargs['segment_ids'].cpu().numpy()),
+                }
+            elif self.language_model_class == "Roberta":
+                input_to_onnx = {
+                    'input_ids': numpy.ascontiguousarray(kwargs['input_ids'].cpu().numpy()),
+                    'attention_mask': numpy.ascontiguousarray(kwargs['padding_mask'].cpu().numpy())
```

Comment:
What is needed if we add a new model here? Just adding the mapping for input names?
If yes, shall we also add XLM-R as it was requested by some users in the community?

Reply:
Yes, just the mapping is needed. However, since XLMRoberta exceeds the default limit of export size in PyTorch, so the parameter `use_external_format` for `convert()` has to be set.